programs/merkle-tree/src/transaction_merkle_tree/initialize_new_merkle_tree_18.rs

```diff
@@ -24,7 +21,30 @@ pub fn process_initialize_new_merkle_tree_18(
     zero_bytes: Vec<[u8; 32]>,
 ) {
     merkle_tree.newest = 1;
+    merkle_tree_authority.transaction_merkle_tree_index += 1;
+
+    initialize_new_merkle_tree(merkle_tree, merkle_tree_authority, height, zero_bytes)
+}
+
+#[cfg(feature = "atomic-transactions")]
+fn initialize_new_merkle_tree(
+    merkle_tree: &mut RefMut<'_, TransactionMerkleTree>,
+    _merkle_tree_authority: &mut Account<'_, MerkleTreeAuthority>,
+    height: usize,
+    _zero_bytes: Vec<[u8; 32]>,
+) {
+    use light_merkle_tree::HashFunction;
+
+    merkle_tree.merkle_tree.init(height, HashFunction::Poseidon)
+}
 
+#[cfg(not(feature = "atomic-transactions"))]
+fn initialize_new_merkle_tree(
+    merkle_tree: &mut RefMut<'_, TransactionMerkleTree>,
```

Comment:
Ah now I understand, it's because you have two different transaction merkle tree pdas now, right?
Wouldn't it be easier to just refactor transaction merkle tree pda so that the layouts are consistent?

Reply:
I have two different transaction Merkle tree layouts, yes.

I feel like aking layouts consistent will be tricky and somehow limiting for the new implementation.

First of all, the old implementation has fields which we don't need anymore:

```rust
    pub pubkey_locked: Pubkey,
    pub time_locked: u64,
    pub lock_duration: u64,
```

Second of all - and this is probably more imporant - for the new implementation, we want to use the separate light-merkle-tree crate, which we have in place for months, and we use it by embedding it as a field:

https://github.com/Lightprotocol/light-protocol/blob/72fcf12d2c61a890121f6f3309c76195d213ac80/programs/merkle-tree/src/event_merkle_tree/initialize.rs#L29

https://github.com/Lightprotocol/light-protocol/blob/72fcf12d2c61a890121f6f3309c76195d213ac80/programs/merkle-tree/src/transaction_merkle_tree/state.rs#L41

To follow your idea with stricly sticking to the old layout, we would basically need to get rid of the light-merkle-tree crate and keep flat struct definitions just in light-merkle-tree-program.

Reply:
couldn't we do the following and add a const generic for padding to the new transaction merkle tree?
```
pub struct TransactionMerkleTree {
        pub merkle_tree_nr: u64,
        pub newest: u8,
        _padding: [u8; 7],
        pub next_queued_index: u64,
        #[feature...]
        pub merkle_tree: MerkleTree<Poseidon, TransactionMerkleTreeConfig>,
        #[not feature]
        pub merkle_tree: OldMerkleTree,
    }
    ```

Reply:
Ah, yes, I could try that

Reply:
It's done, we unified the layout